### ISSUE TITLE
adguard adblocker not working for sonyliv app in android

### ISSUE THREAD

**Issue by spidy2356** [AdguardTeam/AdguardForAndroid#4296]
### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [ yes] I am running the latest version
- [ yes] I checked the documentation and found no answer
- [ yes] I checked to make sure that this issue has not already been filed

### Issue Details
after enabling adguard adblocker android, sonyliv app content streaming not working, it freezes or remains buffering

- **AdGuard version:**
  - v4.0 nightly 35
- **Filtering mode:**
  - local vpn
- **Device:**
  - poco x2
- **Operating system and version:**
  - android 11
- **Root access:**
  - no

### Expected Behavior
sonyliv app should start content streaming without ad.

### Actual Behavior
sonyliv app content not stream, it freezes

### Screenshots

sonyliv app doesn't allow to take screenshot. sorry for that
<details><summary>Screenshot:</summary>

</details>

### Additional Information

almost 1 yr ago I found a filter rule '@@||sonyliv.com^$generichide', after use it the problem was solved for many days but now after the sonyliv's recent update the filter rule not work anymore

**Comment by maxikuzmin** [comment]
@spidy2356 try disabling DNS filtering and Stealth Mode one by one, check if it solved the problem. If everything remains the same, try disabling all filters and repeat the steps, if after disabling all filters the problem is solved, start enabling filters one by one and log into the application to see which filter is interfering

**Comment by spidy2356** [comment]
@maxikuzmin I found the problem. this is occur because of adguard ad filter 'adguard base filter(2.2.59.44)'. when I off it the sonyliv app stream very well. I have a question if I turn off this filter then the ad blockeing on other apps and website will be affected???????

**Comment by spidy2356** [comment]
@maxikuzmin I don't understand what the matter, when I was testing due to close and open of sonyliv app again, and again it works just turn off the ad filters. But after sometimes when I want to use it, it doesn't stream anymore, just buffering. But if I pause AdGuard and open sonyliv app and then play any video and then close the app, resume AdGuard, then it works finely. What the matter I don't understand.........please help sir

**Comment by maxikuzmin** [comment]
@spidy2356 is streaming Sonyliv app content possible only if all filters are disabled? Try disabling the filters one by one, I still think the problem is not all filters, but only one.
If I disable just the base filter, will Sonyliv still work? I recommend that you restart Sonyliv each time you disable the filter.
"But if I pause AdGuard and open sonyliv app and then play any video and then close the app, resume AdGuard, then it works finely." - I think the content may have been cached, which is why it worked.

**Comment by spidy2356** [comment]
@maxikuzmin first time when I testing, I disable all filter except ad filters and turn off the ad filters one by one and testing sonyliv app, so due to close and open of sonyliv app it start working. that is why yesterday fisrt time I say about only ad filters problem. but 2nd time when I wnt to use it I observe that after turning off all filters the app doesn't work. that's why I say the second method 



> @maxikuzmin I don't understand what the matter, when I was testing due to close and open of sonyliv app again, and again it works just turn off the ad filters. But after sometimes when I want to use it, it doesn't stream anymore, just buffering. But if I pause AdGuard and open sonyliv app and then play any video and then close the app, resume AdGuard, then it works finely. What the matter I don't understand.........please help sir

but if I use adaway the works finely, so plz tell me what filters adaway use, then I will only use those filters.
I think this is not the problem of filters, if sonyliv app detects ad block at first then it block streaming but if I pause adblock and open app and then close app and turn on adblock then it works( with all filters), so what the problem

**Comment by maxikuzmin** [comment]
@spidy2356 what's the adaway? You didn't mention it before

**Comment by spidy2356** [comment]
> @spidy2356 what's the adaway? You didn't mention it before

Adaway is open source Android adblocker. Sorry that's my mistake I forgot

**Comment by maxikuzmin** [comment]
@spidy2356 when testing do you have both blockers enabled or just one? Please try closing Adaway completely and repeat the steps. 
Did I understand correctly that if you disable all filters, the app still doesn't show the video unless you restart it? In doing so, are you trying to watch different videos?

**Comment by spidy2356** [comment]
> @spidy2356 when testing do you have both blockers enabled or just one? Please try closing Adaway completely and repeat the steps. Did I understand correctly that if you disable all filters, the app still doesn't show the video unless you restart it? In doing so, are you trying to watch different videos?

@maxikuzmin when I was testing, I just enable one ad blocker that is AdGuard. AdGuard and adaway doesn't work at same time cause both work via VPN based filtering. You can check adaway details from here-
https://github.com/AdAway/AdAway
and second answer is, yes you understand correctly, the sonyliv app doesn't work after disable all filters ( some time works due to again and again close and opening of sonyliv app), it works only when I pause adguard then open sonyliv app and play a video  (shows ad) then close the app properly and resume adguard then the adblocking in app and content steming both work smoothly.
                                                       
before report here I have read many forums on google about this issue, I haven't found any solutons yet. few says it works in android tv after change the dns but the forums about changing dns is very old. so, I suggest you to check it out personally, after that you may found any solutions,i think............thank you

**Comment by maxikuzmin** [comment]
@spidy2356 to troubleshoot this issue, we need to get additional logs.

Here's what we need you to do:

- Collect the debug log as it's [explained here](https://kb.adguard.com/en/android/solving-problems/log#debug);
- Remember the exact time when the issue was reproduced. We will need it to find the corresponding records in the log file;
- Additionally, capture the system bug report after reproducing the issue. [Here's the instruction](https://developer.android.com/studio/debug/bug-report);
- Send both files to devteam@adguard.com. Mention the Github issue number and the exact time when the issue was reproduced.

**Comment by spidy2356** [comment]
@maxikuzmin you said remember the exact time that means clock time(IST)????
and how to get poco x2 system bug report????........I read the forum but nothing help, in my device there is a 'service and feedback' app where I can report a bug but how to get capture system bug report???

**Comment by maxikuzmin** [comment]
@spidy2356 I attached the instructions in a previous [post](https://kb.adguard.com/en/android/solving-problems/log#debug). Did you read it? You should have the same menu as the other android phones
What don't you understand? Let's figure it out together, I'll help you

**Comment by spidy2356** [comment]
@maxikuzmin yeah. I read the post but in poco X2 in was little different. I find it from YouTube and send the files on that day, what you need, in the email, you said. Thank you...I believe you find the issue soon

**Comment by maxikuzmin** [comment]
@spidy2356 sorry, I don't see logs in the mail. Did you send to [devteam@adguard.com](mailto:devteam@adguard.com)? Please resend, don't forget to include the time of the problem

**Comment by spidy2356** [comment]
> @spidy2356 sorry, I don't see logs in the mail. Did you send to [devteam@adguard.com](mailto:devteam@adguard.com)? Please resend, don't forget to include the time of the problem

@maxikuzmin I send it 6 days before, here is the screenshot
![Screenshot_2022-12-30-15-13-03-074_com google android gm](https://user-images.githubusercontent.com/101243689/210056488-495c27ce-0ff4-4f2d-a72d-4ab1c78956aa.jpg)
Do I resent the same email or sent new log file??

**Comment by maxikuzmin** [comment]
@spidy2356 I found. We're looking into them.

**Comment by spidy2356** [comment]
> @spidy2356 I found. We're looking into them.

@maxikuzmin  Great job bro....thanks for helping me ☺️

**Comment by maxikuzmin** [comment]
@maxikuzmin I hope we find what the problem is. I'll text you later

**Comment by maxikuzmin** [comment]
@spidy2356 can I ask you to also send debug logs with filters disabled?

**Comment by spidy2356** [comment]
@maxikuzmin should I run sonyliv, and tell exact time???......can I send it here????

**Comment by spidy2356** [comment]
[adguard_logs_4.0.79_1201_162746.zip](https://github.com/AdguardTeam/AdguardForAndroid/files/10400886/adguard_logs_4.0.79_1201_162746.zip)
here is the file, when all filters are turned off and sonyliv app streaming working.
the exact time when I open and play a video is 4:27 PM

**Comment by maxikuzmin** [comment]
@spidy2356 yes, that's right. Repeat the same steps, just with the filters off. You can attach it here

**Comment by spidy2356** [comment]
> @spidy2356 yes, that's right. Repeat the same steps, just with the filters off. You can attach it here

@maxikuzmin I already send it here https://github.com/AdguardTeam/AdguardForAndroid/issues/4296#issuecomment-1380158825

**Comment by maxikuzmin** [comment]
@spidy2356 thank you! Will be back with an answer later

**Comment by spidy2356** [comment]
@maxikuzmin I have a question, is there any way to block ads in apps of bluestacks 5 nougat 64 bit?????
if you know any way plz tell me

**Comment by maxikuzmin** [comment]
@spidy2356 for PC? I think you need to create new issue for this issue

**Comment by spidy2356** [comment]
@maxikuzmin I tested this app today once again. Adguard uses local vpn for adblocking that is the reason for streaming. I stream many types of videos for a long time and some times the app is noticed me 'slow connection' and few videos playbacks after waiting some times and app is noticed me "internet connection is detected"...so I think due to any reason the app doesn't detect internet connection when ad blocker is running....you should check it....and I apply for beta tester...how to get selected and get a free beta version??

**Comment by maxikuzmin** [comment]
@spidy2356 I think in this case it would be nice  to also take screenshots of the filtering log and send to devteam@adguard.com. 
Maybe you have a second device to take pictures of the ads you see? It would be nice for us to see exactly what's showing in the app, including JioTV
[Here](https://adguard.com/en/beta.html) you will find all the information about beta testing

**Comment by spidy2356** [comment]
> @spidy2356 I think in this case it would be nice to also take screenshots of the filtering log and send to [devteam@adguard.com](mailto:devteam@adguard.com). Maybe you have a second device to take pictures of the ads you see? It would be nice for us to see exactly what's showing in the app, including JioTV [Here](https://adguard.com/en/beta.html) you will find all the information about beta testing

@maxikuzmin currently I don't have any extra phone, so you must be wait for two days so that I manage a phone then I send you ss.....ok. and how to manage to get full filterlog .....plz help because taking screenshot is most to impossible

**Comment by maxikuzmin** [comment]
@spidy2356 close all apps, clear filtering log and repeat steps to retry ads. You can send multiple screenshots at once.

**Comment by maxikuzmin** [comment]
@spidy2356 any news? I have only seen screenshots from JIOTv

**Comment by spidy2356** [comment]
@maxikuzmin i am sending sonyliv app filterlog and app ss........do you want in app video????

**Comment by spidy2356** [comment]
> @spidy2356 any news? I have only seen screenshots from JIOTv

@maxikuzmin I don't send anything about Jio TV , because it doesn't show ads anymore......and I suggest closing the jiotv issue....because Jio products have already lots of bugs, so I don't think it is fully AdGuard problem......so drop it for now

**Comment by spidy2356** [comment]
@maxikuzmin I have a question........firefox android doesn't support HTTPS filtering of adguard.???.......because when HTTPS filtering is enable for firefox, most of the websites and site redirects doesn't work or error to load......why??

**Comment by maxikuzmin** [comment]
@spidy2356 HTTPS filtering for Firefox is supported. Check if the AdGuard certificate is displayed in the Firefox app. I also recommend reading this [article](https://kb.adguard.com/en/android/solving-problems/firefox-certificates), it should help you

**Comment by maxikuzmin** [comment]
@spidy2356 try adding to exclusions [sonyliv-td-static.akamaized.net](http://sonyliv-td-static.akamaized.net/)

**Comment by spidy2356** [comment]
How to do that??

**Comment by spidy2356** [comment]
> @spidy2356 try adding to exclusions [sonyliv-td-static.akamaized.net](http://sonyliv-td-static.akamaized.net/)

@maxikuzmin I add this thing to exclusion and also block Google ads services from filter logs and also unblock one thing from filter log but unfortunately I didn't recall the name. And the videos are now stream properly....I wait for two days for a proper decision. Thank you
(If you know how to see changed filter rule, then I can tell you what I unblock)

**Comment by maxikuzmin** [comment]
@spidy2356 if you blocked something from filtering log, it should be in Ad blocking -> User rules

**Comment by spidy2356** [comment]
@maxikuzmin I didn't found anything in user rules section. But what I do in filter log that didn't work at all.....I think if the app doesn't run ads then it block internet connection....can you fix it?

**Comment by maxikuzmin** [comment]
@spidy2356 what do you mean? If your internet connection is blocked, then you can't see any content, not even ads

**Comment by spidy2356** [comment]
@maxikuzmin yes that's all....I told you in mail...in app the streaming is pause like pic
I pause adguard then play video.... after that I enable adguard... as a result I saw one ad the adguard start blocking ads and streaming also possible.....If I change video sometimes app says internet connection is slow..but I have broadband connection.

So what you think??

**Comment by maxikuzmin** [comment]
@spidy2356 sorry, you wrote that "videos are now streaming properly".
So after adding to exclusions [sonyliv-td-static.akamaized.net](http://sonyliv-td-static.akamaized.net/) it didn't help? Does the problem also recur? If the protection is off, there are no problems with the Internet when you use sonyliv?

**Comment by spidy2356** [comment]
@maxikuzmin no, the problem doesn't recur at all after adding to exclusion sonyliv-td-static.akamaized.net
And yeah, there is no internet issue when adguard protection is off.
If I just open the app turning on the adguard protection then internet connection are just closed for app...nothing plays at all...but If I puse protection and open sonyliv and play a video, then just turn on the protection then the app start playing video with ads but after one ad the video just playing but the ads are filtered by adguard.
When this whole process occur once then the app working with protect for 2..3 hrs..after that...I have to do the whole process once again to play video without ads.
If You didn't understand should I make a video and mail you using pcloud??

**Comment by maxikuzmin** [comment]
@spidy2356 initially we had a problem with the presence of ads in sonyliv. Now with the exclusion of _sonyliv-td-static.akamaized.net_ you have no advertising, but you lose Internet?
Please describe briefly, what problems are you having in the app right now?
Also please send new debug logs

**Comment by spidy2356** [comment]
@maxikuzmin the issue in sonyliv app is same, nothing changes at all....I think I didn't understand it well before so that's why I didn't understand you well........so forget about everything.....the main thing is when If I just open the app turning on the adguard protection then internet connection are just closed for app.......cause nothing streams at all....and when my internet connection is turned off then the same situation occur......so what you think by my opinion........isn't the app block internet when I try to block ads???......and other observation is......if i pause protection and play any one video and then turn on adguard protection then video start streaming after 1 ad...but if I change to another video, then the app says 'slow internet connection', although I'm connected with stable broadband connection....so what you think???.......if you didn;t understand then next time I will make a video and send you via pcloud....thank you

**Comment by maxikuzmin** [comment]
@spidy2356 so we have two cases:
1. When with protection enabled, the Internet in sonyliv is blocked
2. When you first go to sonyliv with protection turned off, start watching the video then turn on adguard protection, the video will start streaming after 1 ad, but if you switch to another video, then the app will say "slow internet connection"

Please repeat situation "1" and unload the debug logs, then repeat situation "2" and also unload the debug logs. We need the debug logs separately for each situation. Please do not forget to specify the playback time. 
Send the dubug logs to [devteam@adguard.com](mailto:devteam@adguard.com)

**Comment by spidy2356** [comment]
@maxikuzmin ok. I will send them in email tonight.....but how to specify playback time, do you mean the clock time when I start a video???.......you can try the app by your own for better understanding...don't you???

**Comment by maxikuzmin** [comment]
@spidy2356 unfortunately I cannot test the Sonyliv app because it is a localized app and this app is not supported in my country. So I ask for more details from you :)
Yes, you need to repeat the problem and remember what time it happened. When you send the logs, specify the time to each logs (to both situations)

**Comment by spidy2356** [comment]
@maxikuzmin I send the mail. And is there no one from India in your developer team?

**Comment by maxikuzmin** [comment]
@spidy2356 no. Thanks, I got the logs, but they are now from version 3.6.11.
Before that, you were sending logs from version 4.0.
So on which version is the problem?

**Comment by spidy2356** [comment]
@maxikuzmin but the problem is the same...the version change because of your beta key

**Comment by maxikuzmin** [comment]
@spidy2356 please unload the logs from version 4.0, I think it would not be superfluous

**Comment by spidy2356** [comment]
@maxikuzmin I uninstall the nightly version to use this beta key and use this 3.6.11 version currently....so what I do now 😢

**Comment by spidy2356** [comment]
@maxikuzmin I tried some premium videos of sonyliv which are few miutes free.....they are works perfectly with adguard protection on. and the app loads well but the issue is free videos doesn't works with adguard protection on....looks like internet access are bloked. so I make a video and provide filterlog with playing all kinds of videos( premium, free) and send in mail in few days. ok.😊

**Comment by maxikuzmin** [comment]
@spidy2356 okay, we're waiting

**Comment by spidy2356** [comment]
@maxikuzmin I send it to the email via pcloud, see the video. Firstly which video play flawlessly this is a premium video and secondly which video play,(not play every time flawlessly this type of category), just one episode free then premium and fully free video doesn't play at all with ad blocker.

**Comment by spidy2356** [comment]
@maxikuzmin is the video understandable or I make a better video and send you, it is little hard to make video without ph holder....what your suggest...make another one or not

**Comment by maxikuzmin** [comment]
@spidy2356 if you can provide more materials to explore the problem, it would not be out of place. I have handed over all the materials, now we're need some time. Send more materials if you can provide more information about the problem

**Comment by spidy2356** [comment]
@maxikuzmin what type of material or information you need?...You can take time as much you need...I just ask you, is the video which I send you, is understandable or not

**Comment by maxikuzmin** [comment]
@spidy2356 sorry for the long wait, we need a little more time

**Comment by spidy2356** [comment]
@maxikuzmin it's ok....I will wait. No problem 🥰

**Comment by spidy2356** [comment]
@maxikuzmin  there is a video streaming platform called atoplay.com, but adguard extension/or any other adblocking extension can block its first 5secs skippable ads, then what can I do and how is it possible?
if you said then I can make this point a new issue.

and secondly, I wanna ask you something, is that possible to block youtube ads by using the accessibility feature of android in Adguard? like I found an automation app "macro droid", and it has a few templates for blocking youtube app ads and it works well and it requires an accessibility feature, so I ask you if it was possible.

**Comment by maxikuzmin** [comment]
@spidy2356 on another app's problem, please start another issue. 
Viewing YouTube without ads is only possible with the "Share" button in YouTube

**Comment by spidy2356** [comment]
@maxikuzmin  ok I will create new issue. And I know the share option to watch YouTube videos without ads...but I'm asking is it possible using accessibility feature of Android like Macrodroid app?? Macrodroid can block official YouTube app ads...just search youtube u get know about it

**Comment by maxikuzmin** [comment]
@spidy2356 surely there are other ways to watch YouTube without ads, but here we are only talking about AdGuard :)
For your main question, it will take some more time

**Comment by spidy2356** [comment]
@maxikuzmin I wanna talk about the issue I registered. Recently I use only private dns> dns.adguard.com from network settings and the sonyliv app don't detect it and video plays without ads also.
So is that possible to make a rule for sonyliv so that only dns will be in use and other filtering not work for sonyliv. If possible then make such a rule then I will try and inform u. Thank you

**Comment by maxikuzmin** [comment]
@spidy2356 so the sonyliv app doesn't detect DNS, and then there are no ads in the app? And how do you check that the app is not detecting DNS?
Then why make a rule to filter the application only for DNS if you say that sonyliv does not detect DNS?
Do you think the filters are interfering and that's why the ads are showing? Try turning off filters manually, leaving only DNS and see how it works

**Comment by spidy2356** [comment]
@maxikuzmin I turn off the adguard app and set private DNS from the android network setting > private DNS > host name: dns.adguard.com. that's it, and this method blocks every app and website ads and on sonyliv video plays ads free without no issue.
I don't know sonyliv is detecting DNS or not but this method occurs no issue during video playback ( like video not plays at all if I use adguard app).
Maybe it's not adguard filtering problem as adguard work as VPN. so sonyliv detects VPN and stops video playback.
I talked to a Bangladeshi person, he said that if he turns on VPN and plays Sonyliv and video not plays. the app detects it, and if he opens Sonyliv and then turns on VPN, it somehow starts working. just like my adguard issue.
so maybe the issue(what I registered) occurs because of VPN filtering.

**Comment by spidy2356** [comment]
@maxikuzmin  I cross-checked, again and again, and only DNS filtering via adguard app occurs with no issue during video play in Sonyliv. then make a rule.
that's the final solution.

**Comment by maxikuzmin** [comment]
@spidy2356 unfortunately there is no way to create a DNS filtering function for just one application. 
But then you can enable DNS for everything and that should work as well, right?

**Comment by spidy2356** [comment]
@maxikuzmin yes, if only dns filtering enable them everything works well.
Is that not possible to make user rule for a specific app so that other filters not work for this app except dns?

**Comment by maxikuzmin** [comment]
@spidy2356 yes, you can turn it off for sonyliv in "App management"  -> "Filter app content"
A rule for not filtering an application @@*$app=package_id 
"package_id" you can find in Settings tab -> Filtering -> Apps -> Apps management -> click on the desired application -> context menu -> Statistics -> context menu -> Technical info -> Package name

**Comment by spidy2356** [comment]
@maxikuzmin it's working fine now. Thanks for ur help.

**Comment by maxikuzmin** [comment]
@spidy2356 this is great news! Do you have any unresolved questions?

**Comment by spidy2356** [comment]
@maxikuzmin  yes I have. When I surf the internet there are some shortlinks which I found and even some website who use anti adblock script. some of then are bypassed by using "block ads on website" option given by adguard extension, some are bypasses by javascript off. but some r not possible to bypass and even I turn off javascript then anti adblock script gone but website got static and I can't go forward by next link or links not work. so what can I do now?

**Comment by maxikuzmin** [comment]
@spidy2356 I think that since this is no longer relevant to the main topic of this issue, please open a new issue and we will try to figure it out. 
I'm glad that we were able to figure out the sonyliv application :)
I'm closing this issue.

**Comment by spidy2356** [comment]
@maxikuzmin yes thik close it. But how to open new issue on this topic because this not for specific website so plz suggest me something what can I do about the issue

**Comment by maxikuzmin** [comment]
@spidy2356 just create a new issue, like you did with this one. Name the topic of the problem, for example "Ads are not blocked on some sites" (If I understand correctly).
Describe the steps of reproducing the problem and attach links to the sites

**Comment by spidy2356** [comment]
@maxikuzmin  the issue occurs mainly for anti adblock script, so how can I issue it to adguard team?? is it ok if I do so?

**Comment by maxikuzmin** [comment]
@spidy2356 you can try creating an issue in the repository https://github.com/AdguardTeam/AdguardFilters if the anti-blocker script blocks filters.
Maybe they can help you in some way
